pkg/resource/replication_group/delta_util.go

```diff
@@ -62,3 +79,74 @@ func engineVersionsMatch(
 
 	return false
 }
+
+// compares desired.Spec to last applied slow log config in latest.Status to determine
+// whether to invoke an update or not. This comes down to only trying to apply a particular configuration
+// once.
+func slowLogRequiresUpdate(desired *resource, latest *resource) bool {
+	var desiredConfig *svcapitypes.LogDeliveryConfigurationRequest
+	var lastRequestedConfig *svcapitypes.LogDeliveryConfiguration
+
+	// retrieve desiredConfig from desired Spec
+	if desired.ko.Spec.LogDeliveryConfigurations != nil {
+		for _, config := range desired.ko.Spec.LogDeliveryConfigurations {
+			if config.LogType != nil && *config.LogType == "slow-log" {
```

Comment:
Instead of hard coding `"slow-log"` You can use the generate enum variables here https://github.com/aws-controllers-k8s/elasticache-controller/blob/main/apis/v1alpha1/enums.go#L87-L91